tool_quality_checks/copy_ogr_layer.py

```diff
@@ -0,0 +1,155 @@
+        )
+        corrected_in_layer = corrected_in_source.GetLayerByName(corrected_layer_name)
+
+        # check projection of input file
+        check_sr = get_projection(in_srid)
+        if check_sr is None:
+            logger.warning(
+                "[!] Warning : Projection is not complete EPSG projection code missing in shapefile."
+            )
+
+        # correct vector layer to solve issues and stuff
+        # correct_in_source, correct_layer_name = in_layer, out_name
+        corrected_in_source, corrected_layer_name = correct_vector_layer(
+            in_layer, out_name, epsg=28992
+        )
+        corrected_in_layer = corrected_in_source.GetLayerByName(corrected_layer_name)
+
+        options = [
+            "OVERWRITE=YES",
+            "SCHEMA={}".format(schema),
+            "SPATIAL_INDEX=GIST",
+            # try to comment row below when running to transaction block error
+            "FID=id",
+            "PRECISION=NO",
+            "GEOMETRY_NAME=geom",
+            "DIM=2",
+        ]
+
+        ogr.RegisterAll()
+        # TODO srid is now based on in_layer, which could be a strange spatial reference
+        # TODO findout how to make the target ref 28992 by default
+        new_layer = out_source.CreateLayer(
+            corrected_layer_name,
+            srs=corrected_in_layer.GetSpatialRef(),
+            geom_type=corrected_in_layer.GetGeomType(),
+            options=options,
+        )
+    else:
+        corrected_layer_name = correct_layer_name_length(out_name)
+        corrected_in_layer = in_layer
+
+        options = [
+            "OVERWRITE=YES",
+            "SCHEMA={}".format(schema),
+            # try to comment row below when running to transaction block error
+            "FID=id",
+            "PRECISION=NO",
+        ]
+
+        new_layer = out_source.CreateLayer(corrected_layer_name, options=options)
+
+    field_names_list = []
+    defn = corrected_in_layer.GetLayerDefn()
+    for i in range(defn.GetFieldCount()):
+        new_layer.CreateField(defn.GetFieldDefn(i))
+        field_names_list.append(defn.GetFieldDefn(i).GetName())
+
+    new_layer.StartTransaction()
+    for j in range(corrected_in_layer.GetFeatureCount()):
+        copy_feature = corrected_in_layer.GetFeature(j)
+        copy_feature.SetFID(-1)
+        new_feature = ogr.Feature(new_layer.GetLayerDefn())
+        new_feature.SetFID(-1)
+        for key in field_names_list:
+            new_feature[key] = copy_feature[key]
+        if has_geom:
+            new_geom = copy_feature.geometry()
+            new_feature.SetGeometry(new_geom)
+        new_layer.CreateFeature(new_feature)
+        if j % 128 == 0:
+            new_layer.CommitTransaction()
+            new_layer.StartTransaction()
+    new_layer.CommitTransaction()
+
+    if new_layer.GetFeatureCount() == 0 and in_layer.GetFeatureCount() > 0:
+        raise ValueError("output feature count is 0, while input is not")
+
+    new_layer = None
```

Comment:
Dit is vast een trucje om iets netjes af te sluiten of om garbage collection aan te zetten ofzo? 
Ik zou er even een kort commentaar achter zetten, want het is typisch zo'n regel die iemand anders netjes opruimt :-)

```suggestion
    new_layer = None  # Used to do xyz.
```